packages/headless/doc-parser/src/function-resolver.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  ApiEntryPoint,
+  ApiFunction,
+  ApiInterface,
+} from '@microsoft/api-extractor-model';
+import {DocComment} from '@microsoft/tsdoc';
+import {findApi} from './api-finder';
+import {ObjEntity} from './entity';
+import {buildEntity, buildFuncEntity} from './entity-builder';
+import {resolveParams} from './function-param-resolver';
+import {resolveInterfaceMembers} from './interface-resolver';
+
+export function resolveFunction(
+  entry: ApiEntryPoint,
+  fn: ApiFunction,
+  shallowParamIndices: number[]
+) {
+  const params = resolveParams(entry, fn, shallowParamIndices);
+  const returnTypeText = fn.returnTypeExcerpt.text;
+  const returnTypeInterface = findApi(entry, returnTypeText) as ApiInterface;
+
+  const returnType = buildObjEntityFromInterface(entry, returnTypeInterface);
+
+  return buildFuncEntity({
+    name: fn.name,
+    comment: (fn.tsdocComment as unknown) as DocComment,
+    params,
+    returnType,
+  });
+}
+
+function buildObjEntityFromInterface(
```

Comment:
`buildObjEntityFromParam` could be simplified if this function was exported. Perhaps it could be moved to `interface-resolver` and exported?

Reply:
I'm not sure hmm, there's a slight nuance with this function in that the `name` and `type` are the same because it is resolving the function return type. e.g.

```
function buildFacet(...): Facet {}
```

This differs from resolving the property or function parameter interface, where the prop/param usually has one name and the type can be a different name.